src/lib.rs

```diff
@@ -977,6 +977,21 @@ pub fn version_is_supported(version: u32) -> bool {
     }
 }
 
+macro_rules! add_frame {
```

Comment:
This probably needs a more decruptive name, like `push_frame_to_pkt` or something like that, also would be good to have a bit of documentation.

Reply:
Changed to `push_frame_to_pkt` and add a comment.